nemo/collections/tts/models/tacotron2.py

```diff
@@ -73,17 +93,16 @@ def __init__(self, cfg: DictConfig, trainer: 'Trainer' = None):
         # Ensure passed cfg is compliant with schema
         try:
             OmegaConf.merge(cfg, schema)
-            self.pad_value = self._cfg.preprocessor.pad_value
+            self.pad_value = cfg.preprocessor.pad_value
         except ConfigAttributeError:
-            self.pad_value = self._cfg.preprocessor.params.pad_value
+            self.pad_value = cfg.preprocessor.params.pad_value
             logging.warning(
                 "Your config is using an old NeMo yaml configuration. Please ensure that the yaml matches the "
                 "current version in the main branch for future compatibility."
             )
-
         self._parser = None
-        self.audio_to_melspec_precessor = instantiate(self._cfg.preprocessor)
-        self.text_embedding = nn.Embedding(len(cfg.labels) + 3, 512)
+        self.audio_to_melspec_precessor = instantiate(cfg.preprocessor, highfreq=cfg.train_ds.dataset.highfreq)
+        self.text_embedding = nn.Embedding(self.num_tokens, 512)
```

Comment:
I still don't understand how bos and eos are handled. Your embedding only has self.num_tokens but you pass in self.num_tokens+1 here: https://github.com/NVIDIA/NeMo/pull/4103/files#diff-ebdd6ab591a0b633b1bea4d397fa6a561a5ee187193d8a0bcb30a605f9ae94c7R135